packages/babel-parser/src/parser/comments.js

```diff
@@ -39,10 +39,11 @@ export default class CommentsParser extends BaseParser {
   }
 
   adjustCommentsAfterTrailingComma(node: Node, elements: Node[]) {
-    if (elements.length === 0) {
+    if (this.state.leadingComments.length === 0) {
       return;
     }
-    if (this.state.leadingComments.length === 0) {
+
+    if (elements.length === 0) {
```

Comment:
I'm surprised by this change. Checking an array's length should be much faster than accessing 3 props.